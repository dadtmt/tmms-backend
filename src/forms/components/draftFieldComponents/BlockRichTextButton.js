import React, { PropTypes } from 'react'
import { Button } from 'react-bootstrap'
import FontAwesome from 'react-fontawesome'

const BlockRichTextButton = ({
  iconName,
  isActive,
  label,
  toggleBlockType
}) => <Button
  active={isActive}
  onClick={toggleBlockType}
  title={label}
>
  <FontAwesome name={iconName} />
</Button>

BlockRichTextButton.propTypes = {
  blockType: PropTypes.string,
  iconName: PropTypes.string,
  isActive: PropTypes.bool,
  label: PropTypes.string,
  toggleBlockType: PropTypes.func
}

export default BlockRichTextButton
