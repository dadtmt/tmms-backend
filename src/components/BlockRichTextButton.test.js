import React from 'react'
import renderer from 'react-test-renderer'

import BlockRichTextButton from './BlockRichTextButton'

it('renders correctly', () => {
  const props = {
    iconName: 'bold',
    isActive: true,
    label: 'bold',
    toggleBlockType: jest.fn()
  }
  expect(
    renderer.create(<BlockRichTextButton {...props} />).toJSON()
  ).toMatchSnapshot()
})
