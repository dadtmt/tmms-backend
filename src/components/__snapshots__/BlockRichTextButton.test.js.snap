exports[`test renders correctly 1`] = `
<button
  className="btn btn-default active"
  disabled={false}
  onClick={[Function]}
  title="bold"
  type="button">
  <span
    aria-hidden={true}
    className="fa fa-bold" />
</button>
`;
